pkg/cmd/kind/test/internal/testcmd/testcmd.go

```diff
@@ -0,0 +1,372 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package testcmd implements generic test command logic
+package testcmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha4"
```

Comment:
shouldn't we use the internal APIs?

Reply:
no. internal apis are for:
- translating between versions
- allowing runtime code implementing the API to handle a single version, possibly with extra (computed?) data that is not part of the public API.

the CLI is a client of the runtime.